plugins/linux/ifplugin/interface_config.go

```diff
@@ -470,18 +470,18 @@ func (plugin *LinuxInterfaceConfigurator) configureLinuxInterface(nsMgmtCtx *nsp
 		plugin.log.Debugf("MTU %d set to interface %s", ifConfig.Mtu, ifConfig.HostIfName)
 	}
 
-	idx := plugin.getLinuxInterfaceIndex(ifConfig.HostIfName)
-	if idx < 0 {
-		return fmt.Errorf("failed to get index of the Linux interface %s", ifConfig.HostIfName)
+	netIf, err := plugin.ifHandler.GetInterfaceByName(ifConfig.HostIfName)
+	if err != nil || netIf == nil || netIf.Index < 0 {
```

Comment:
I'm not sure why you've added all the extra checks that have not been there before. They are completely unnecessary, because: 

* It's not possible to get `nil` interface if `err == nil`, because `net.InterfaceByName` returns pointer to `net.Interface`: https://golang.org/src/net/interface.go#L166
* The `Index` of `net.Interface` is clearly defined that it starts from 0

Just check the error, otherwise assume the interface is valid.